cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -714,6 +717,37 @@ protected static Volume createSecretVolume(String name, String secretName, boole
         return volume;
     }
 
+    protected static Volume createSecretVolume(String name, String secretName, Map<String, String> items, boolean isOpenshift) {
+        int mode = 0444;
+        if (isOpenshift) {
+            mode = 0440;
+        }
+
+        List<KeyToPath> keysPaths = new ArrayList<>();
+
+        for (Map.Entry<String, String> item : items.entrySet()) {
+            KeyToPath keyPath = new KeyToPathBuilder()
+                    .withNewKey(item.getKey())
+                    .withNewPath(item.getValue())
+                    .build();
+
+            keysPaths.add(keyPath);
+        }
+
+        SecretVolumeSource secretVolumeSource = new SecretVolumeSourceBuilder()
+                .withDefaultMode(mode)
+                .withSecretName(secretName)
+                .withItems(keysPaths)
+                .build();
+
+        Volume volume = new VolumeBuilder()
+                .withName(name)
+                .withSecret(secretVolumeSource)
+                .build();
+        log.trace("Created secret Volume named '{}' with source secret '{}'", name, secretName);
+        return volume;
+    }
+
```

Comment:
TBH honest, these are used all over the place. I can move them in this PR. But it will make it just harder to review as it will probably add changes to lot of new files. What about doing it in separate PR later?

Reply:
Fine with it.

Reply:
I opened #2373 for it to have it tracked.